commands/deploy_test.go

```diff
@@ -42,6 +49,152 @@ func Test_deploy(t *testing.T) {
 	}
 }
 
+func Test_overrideFlag_deployToNamespace(t *testing.T) {
+	var request types.FunctionDeployment
+	const defaultNamespace = "openfaas-fn"
+	var cliArgs []string
+	envMap := make(map[string]string)
+	function := make(map[string]stack.Function)
+
+	tests := []struct {
```

Comment:
You should simply be able to unit-test the method that decides over priority when given a number of strings. That's how we did it for the gateway URL. A full end-to-end test of what's sent to the server seems unnecessary to me.

Reply:
Take a look at https://github.com/openfaas/faas-cli/blob/5b36c50280d961d3aa248bd17f901f4ef2774447/commands/priority_test.go

Reply:
Alternative example: https://github.com/openfaas/faas-cli/blob/5b36c50280d961d3aa248bd17f901f4ef2774447/commands/config_priority_test.go